### DISCUSSION TITLE
avrSWAP(79) is defaulted to 0 in the latest ROSCO versions

### DISCUSSION THREAD

**Discussion by ghuwart** [NatLabRockies/ROSCO#232] [Q&A] [answered]
Hello everyone,

I recently upgraded my workflow and made some tests with the latest ROSCO versions (2.6.0 and 2.7.0). Note that I am using my own aero-hydro-servo-elastic tool which is not OPENFAST.

I am having trouble with the fact that avrSWAP(79), the request for loads is now set to zero by default in _ReadSetParameters.f90_:
_avrSWAP(79) = 0.0 ! Request for loads: 0=none_

It seems to be ignored in OPENFAST's ServoDyn v3.4.1:
_!> * Record 79, the request for loads, is ignored; instead, the blade, hub, and yaw bearing loads are always passed to the DLL as if Record 79 was set to 4_
But my code abides by the Bladed API (see Appendix A.1, note 18) and puts a lot of outputs to zero.

![image](https://user-images.githubusercontent.com/128142467/225882592-22b2ad1e-0a32-4f17-b401-42edfb5be458.png)

However I wish to have outputs such as avrSWAP(83), the nacelle nodding acceleration, that is also used for the floating feedback.

I can compile my own ROSCO with avrSWAP(79) not defaulted to zero but I was wondering if there is a reason for putting this value to zero in the first place? I fear that all the codes that are compliant with this point are affected by this initialization.

Thank you.

**Comment by dzalkind** [comment, marked as answer]
I think we ignore `avrSWAP(79)` because ServoDyn ignores that index and sends all the information over the avrSWAP.  Nothing should change in ROSCO if `avrSWAP(79) = 4`.  Since we're getting all the loads in ROSCO, perhaps it should be that way technically.  I can include it in a future bug-fix PR.

**Comment by ghuwart** [comment]
Thank you for the quick reply. I think it is a good idea to either have `avrSWAP(79) = 4`  or no default value at all so that it does not interfere with the code calling ROSCO. This quick fix does the trick in my case and I guess it can help others with similar codes.
